packages/react/src/dialog/root/useDialogRoot.ts

```diff
@@ -46,21 +51,34 @@ export function useDialogRoot(parameters: useDialogRoot.Parameters): useDialogRo
 
   const { mounted, setMounted, transitionStatus } = useTransitionStatus(open);
 
-  const setOpen = useEventCallback((nextOpen: boolean, event?: Event) => {
-    onOpenChange?.(nextOpen, event);
-    setOpenUnwrapped(nextOpen);
-  });
+  const setOpen = useEventCallback(
+    (nextOpen: boolean, event: Event | undefined, reason: OpenChangeReason | undefined) => {
+      onOpenChangeParameter?.(nextOpen, event, reason);
+      setOpenUnwrapped(nextOpen);
+    },
+  );
 
   useAfterExitAnimation({
     open,
     animatedElementRef: popupRef,
     onFinished: () => setMounted(false),
   });
 
+  const handleFloatingUIOpenChange = React.useCallback(
+    (
+      nextOpen: boolean,
+      event: Event | undefined,
+      reason: FloatingUIOpenChangeReason | undefined,
+    ) => {
+      setOpen(nextOpen, event, translateOpenChangeReason(reason));
+    },
+    [setOpen],
+  );
```

Comment:
Another here, though this one (and the one above) actually doesn't need to be wrapped, since it's done internally already